content/blog/2023/05/03/2023-05-03-android-and-jenkins-releases.adoc

```diff
@@ -0,0 +1,319 @@
+For the purposes of this article, let's assume it's okay.
+
+=== Prerequisites
+
+To automate the release process, we need to determine the criteria for a version number, how to update the version number, and what constitutes a release.
+We can use the "link:https://github.com/dipien/semantic-version-gradle-plugin[Semantic Version]" Gradle plugin, which has a strict set of rules to guide us.
+This plugin allows us to link:https://github.com/dipien/semantic-version-gradle-plugin#incrementing-the-project-version[increment] the patch, minor, or major version using Gradle commands.
+We can also use link:https://github.com/dipien/semantic-version-gradle-plugin#version-classifier[classifiers] such as snapshot, beta, alpha, or any other version classifier to define a version name.
+
+[source,groovy]
+----
+version = "1.1.11"
+
+apply plugin: "com.dipien.android.semantic-version"
+----
+
+I then searched for a Jenkins plugin that would create a GitHub release.
+As the saying goes,
+
+[quote]
+There's a plugin for that
+
+but unfortunately, I couldn't find one that meets my needs.
+While there is a plugin called link:https://plugins.jenkins.io/git-changelog/[Git Changelog] that can merge commit messages to produce a readable version of the changes, it doesn't create the release.
+
+=== GitHub release
+
+If you want to stay on the Jenkins side, there isn't a plugin this time.
+However, there are various ways to create a release.
+You can use the link:https://docs.github.com/en/rest?apiVersion=2022-11-28[GitHub REST API] or the link:https://cli.github.com/[`gh`] command, which can handle all the heavy lifting for us.
+Therefore, let's go back to the drawing board and add the command to our Docker image.
+
+[source,dockerfile]
+----
+# Install GitHub command line tool
+ENV GITHUB_TOKEN $GITHUB_TOKEN
+RUN curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg && \
+    chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg && \
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | tee /etc/apt/sources.list.d/github-cli.list > /dev/null && \
+    apt update && apt install -y --no-install-recommends gh
+----
+
+Once that's done, we need to use link:https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/about-authentication-with-a-github-app[GitHub App authentication] to enable `gh` to use our credentials.
+To do this, we have to install the link:https://plugins.jenkins.io/github-branch-source/[GitHub Branch Source plugin] and then create a link:https://www.jenkins.io/blog/2020/04/16/github-app-authentication/[GitHub Application].
+
+No need to replicate here the link:https://github.com/jenkinsci/github-branch-source-plugin/blob/master/docs/github-app.adoc[efficient documentation] existing on GitHub.
```

Comment:
```suggestion
No need to replicate here the link:https://github.com/jenkinsci/github-branch-source-plugin/blob/master/docs/github-app.adoc[efficient documentation] existing on GitHub.
```
I wasn't sure how to rephrase this, as I wasn't sure if the intention is to say "no need to recreate the full documentation" or "the github documentation is accurate and should be copied over"

If you could provide any insight on this, I'd appreciate it very much!

Reply:
Thanks Kevin. 😊 
What I wanted to say is the documentation on GitHub is spot on, a link to it should be enough. 🤷

Reply:
Ahh okay that makes sense.